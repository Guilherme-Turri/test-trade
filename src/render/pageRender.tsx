import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import store from '../storage/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

export const pageRender = (children: React.ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>);
};