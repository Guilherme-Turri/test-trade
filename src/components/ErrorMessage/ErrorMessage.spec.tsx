import { componentRender } from '../../render/componentRender'
import { screen } from '@testing-library/react'
import { ErrorMessage } from './'

describe('<ErrorMessage />', () => {
  it('should render on screen properly', () => {
    const { container } = componentRender(<ErrorMessage>ErrorMsg</ErrorMessage>)
    const msg = screen.getByText('ErrorMsg')
    expect(msg).toBeInTheDocument();
    expect(container).toMatchSnapshot();

  })
})