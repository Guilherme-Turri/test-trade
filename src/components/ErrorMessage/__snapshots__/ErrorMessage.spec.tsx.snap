// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> should render on screen properly 1`] = `
.c0 {
  color: #ff5555;
  font-size: 12px;
}

<div>
  <p
    class="c0"
  >
    ErrorMsg
  </p>
</div>
`;
